libs/components/src/lib/conditions-card/conditions-card.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactElement, useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+
+import { CardWrapper, NewButton, NewCard } from './styles';
+
+import 'antd/dist/antd.css';
+
+export function ConditionsCard(): ReactElement {
+  const navigate = useNavigate();
+  const [dis, setDisabled] = useState(false);
+  const card = React.useRef<HTMLDivElement>(null);
+  const end = React.useRef<HTMLDivElement>(null);
+  const { t } = useTranslation();
+  const isInViewport2 = useIsInViewport();
+  useEffect(() => {
+    setDisabled(!isInViewport2);
+  }, [isInViewport2]);
+
+  function useIsInViewport() {
+    const [isIntersecting, setIsIntersecting] = useState(false);
```

Comment:
this line should return error, you setup state after some section happened. even if its magically work, all hooks should be on the top of the component

Reply:
fixed